src/chunk_index.c

```diff
@@ -578,6 +590,36 @@ chunk_index_delete(Chunk *chunk, Oid chunk_indexrelid, bool drop_index)
 						  scankey, 2, chunk_index_tuple_delete, &drop_index);
 }
 
+static bool
+chunk_index_tuple_found(TupleInfo *ti, void * const data)
+{
+	ChunkIndexMapping * const cim = data;
+	chunk_index_mapping_from_tuple(ti, cim);
+	return false;
+}
+
+
+static ChunkIndexMapping *
+chunk_index_get_by_indexrelid(Chunk *chunk, Oid chunk_indexrelid)
+{
+	ScanKeyData scankey[2];
+	const char *indexname = get_rel_name(chunk_indexrelid);
+	ChunkIndexMapping *cim = palloc(sizeof(ChunkIndexMapping));
+
+	ScanKeyInit(&scankey[0],
+				Anum_chunk_index_chunk_id_index_name_idx_chunk_id,
+				BTEqualStrategyNumber, F_INT4EQ, Int32GetDatum(chunk->fd.id));
+	ScanKeyInit(&scankey[1],
+				Anum_chunk_index_chunk_id_index_name_idx_index_name,
+				BTEqualStrategyNumber, F_NAMEEQ, CStringGetDatum(indexname));
```

Comment:
I think you pointed out this CString vs Name type mismatch. I don't think it is a problem in reality, but we should perhaps do `DirectFunctionCall1(namein, CStringGetDatum(indexname))` and also later fix the other places.

Reply:
done